### PR TITLE
Reorganize zIPL rules

### PR DESCRIPTION
**Pull request by yuumasato**
#### Description:

-  Separate from the zIPL rules what is exclusive to zIPL, which is:
    - Ensure bootmap is up to date (with rule, check and fix)
    - Ensure zIPL is using BLS entries only (with rule and check)
        - IMO, providing automated remediation for this is tricky and risky.
-  With this PR, the zIPL argument rules are much simpler now.

#### Rationale:
- Separate common aspects to its own rules

### CODE REVIEW

**Comment by mildas** [comment]
**Changes identified:**
Rule zipl_bls_entries_only:
 OVAL check is newly added.
Rule zipl_bootmap_is_up_to_date:
 OVAL check is newly added.
Profile ospp on rhel8:
 Rule zipl_bootmap_is_up_to_date, zipl_bls_entries_only added to ospp profile.
Profile cui on rhel8:
 CUI profile extends changed OSPP profile.
Profile stig on rhel8:
 STIG profile extends changed OSPP profile.
Profile rhelh-stig on rhel8:
 RHELH-STIG profile extends changed STIG profile.
Profile rhelh-vpp on rhel8:
 RHELH-VPP profile extends changed OSPP profile.
Profile ospp-mls on rhel8:
 OSPP-MLS profile extends changed OSPP profile.

**Recommended tests to execute:**
 build_product rhel8
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using bash --datastream build/ssg-rhel8-ds.xml zipl_bls_entries_only
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using bash --datastream build/ssg-rhel8-ds.xml zipl_bootmap_is_up_to_date
 tests/test_suite.py profile --libvirt qemu:///system test-suite-vm --datastream build/ssg-rhel8-ds.xml rhelh-vpp
 tests/test_suite.py profile --libvirt qemu:///system test-suite-vm --datastream build/ssg-rhel8-ds.xml ospp
 tests/test_suite.py profile --libvirt qemu:///system test-suite-vm --datastream build/ssg-rhel8-ds.xml rhelh-stig
 tests/test_suite.py profile --libvirt qemu:///system test-suite-vm --datastream build/ssg-rhel8-ds.xml cui
 tests/test_suite.py profile --libvirt qemu:///system test-suite-vm --datastream build/ssg-rhel8-ds.xml ospp-mls
 tests/test_suite.py profile --libvirt qemu:///system test-suite-vm --datastream build/ssg-rhel8-ds.xml stig

**Comment by mildas** [comment]
**Changes identified:**
Rule zipl_bls_entries_only:
 OVAL check is newly added.
Rule zipl_bootmap_is_up_to_date:
 OVAL check is newly added.
 Ansible remediation newly added.
 Bash remediation is newly added.
Profile ospp on rhel8:
 Rule zipl_bootmap_is_up_to_date, zipl_bls_entries_only added to ospp profile.
Profile cui on rhel8:
 CUI profile extends changed OSPP profile.
Profile stig on rhel8:
 STIG profile extends changed OSPP profile.
Profile rhelh-stig on rhel8:
 RHELH-STIG profile extends changed STIG profile.
Profile rhelh-vpp on rhel8:
 RHELH-VPP profile extends changed OSPP profile.
Profile ospp-mls on rhel8:
 OSPP-MLS profile extends changed OSPP profile.

**Recommended tests to execute:**
 build_product rhel8
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using bash --datastream build/ssg-rhel8-ds.xml zipl_bls_entries_only
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using ansible --datastream build/ssg-rhel8-ds.xml zipl_bootmap_is_up_to_date
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using bash --datastream build/ssg-rhel8-ds.xml zipl_bootmap_is_up_to_date
 tests/test_suite.py profile --libvirt qemu:///system test-suite-vm --datastream build/ssg-rhel8-ds.xml ospp
 tests/test_suite.py profile --libvirt qemu:///system test-suite-vm --datastream build/ssg-rhel8-ds.xml stig
 tests/test_suite.py profile --libvirt qemu:///system test-suite-vm --datastream build/ssg-rhel8-ds.xml cui
 tests/test_suite.py profile --libvirt qemu:///system test-suite-vm --datastream build/ssg-rhel8-ds.xml rhelh-stig
 tests/test_suite.py profile --libvirt qemu:///system test-suite-vm --datastream build/ssg-rhel8-ds.xml rhelh-vpp
 tests/test_suite.py profile --libvirt qemu:///system test-suite-vm --datastream build/ssg-rhel8-ds.xml ospp-mls

File: linux_os/guide/system/bootloader-zipl/zipl_bootmap_is_up_to_date/ansible/shared.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+      stat:
+        path: /boot/bootmap
+      register: boot_bootmap
+
+    - name: "Obtain stats of /etc/zipl.conf"
+      stat:
+        path: /etc/zipl.conf
+      register: zipl_conf
+
+    # TODO: handle /boot/loader/entries/*.conf
```
Comment (yuumasato) [inline linux_os/guide/system/bootloader-zipl/zipl_bootmap_is_up_to_date/ansible/shared.yml:19]: I don't see an easy way to do this, leaving as a TODO item for the future.

File: linux_os/guide/system/bootloader-zipl/zipl_bootmap_is_up_to_date/rule.yml [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    Every time a boot entry or zIPL configuration is changed <tt>/boot/bootmap</tt> needs to
+    be updated to reflect the changes.<br />
+    Run <tt>zipl</tt> command to generate an updated <tt>/boot/bootmap</tt>.
+
+rationale: |-
+    The file <tt>/boot/bootmap</tt> contains all boot data, keeping it up to date is crucial to
+    boot correct kernel and options.
+
+severity: medium
+
```
Comment (vojtapolasek) [inline linux_os/guide/system/bootloader-zipl/zipl_bootmap_is_up_to_date/rule.yml:18]: missing CCE

File: linux_os/guide/system/bootloader-zipl/zipl_bootmap_is_up_to_date/bash/shared.sh [hunk 3]
```diff
@@ -0,0 +1,3 @@
+# platform = Red Hat Enterprise Linux 8
+
+/usr/bin/zipl
```
Comment (vojtapolasek) [inline linux_os/guide/system/bootloader-zipl/zipl_bootmap_is_up_to_date/bash/shared.sh:3]:
```suggestion
/usr/sbin/zipl
```

File: linux_os/guide/system/bootloader-zipl/zipl_bls_entries_only/oval/shared.xml [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+  <ind:textfilecontent54_test id="test_zipl_bls_entries_only"
+  comment="Test presence of image configuration in /etc/zipl.conf"
+  check="all" check_existence="none_exist" version="1">
+    <ind:object object_ref="object_zipl_bls_entries_only" />
+  </ind:textfilecontent54_test>
+
+  <ind:textfilecontent54_object id="object_zipl_bls_entries_only"
+  version="1">
+    <ind:filepath operation="pattern match">^/etc/zipl.conf$</ind:filepath>
+    <ind:pattern operation="pattern match">^image\s*=.*$</ind:pattern>
```
Comment (matejak) [inline linux_os/guide/system/bootloader-zipl/zipl_bls_entries_only/oval/shared.xml:23]: I found this example: https://insights-core.readthedocs.io/en/latest/shared_parsers_catalog/zipl_conf.html It indicates that there may be leading whitespaces before `image`.

**Comment by mildas** [comment]
**Changes identified:**
Rule zipl_bls_entries_only:
 OVAL check is newly added.
Rule zipl_bootmap_is_up_to_date:
 Ansible remediation newly added.
 Bash remediation is newly added.
 OVAL check is newly added.
Profile ospp on rhel8:
 Rule zipl_bootmap_is_up_to_date, zipl_bls_entries_only added to ospp profile.
Profile cui on rhel8:
 CUI profile extends changed OSPP profile.
Profile stig on rhel8:
 STIG profile extends changed OSPP profile.
Profile rhelh-stig on rhel8:
 RHELH-STIG profile extends changed STIG profile.
Profile rhelh-vpp on rhel8:
 RHELH-VPP profile extends changed OSPP profile.
Profile ospp-mls on rhel8:
 OSPP-MLS profile extends changed OSPP profile.
Profile ospp on tests:
 Rule zipl_bootmap_is_up_to_date, zipl_bls_entries_only added to ospp profile.

**Recommended tests to execute:**
 build_product tests
 tests/test_suite.py profile --libvirt qemu:///system test-suite-vm --datastream build/ssg-tests-ds.xml ospp
 build_product rhel8
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using bash --datastream build/ssg-rhel8-ds.xml zipl_bls_entries_only
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using ansible --datastream build/ssg-rhel8-ds.xml zipl_bootmap_is_up_to_date
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using bash --datastream build/ssg-rhel8-ds.xml zipl_bootmap_is_up_to_date
 tests/test_suite.py profile --libvirt qemu:///system test-suite-vm --datastream build/ssg-rhel8-ds.xml stig
 tests/test_suite.py profile --libvirt qemu:///system test-suite-vm --datastream build/ssg-rhel8-ds.xml ospp-mls
 tests/test_suite.py profile --libvirt qemu:///system test-suite-vm --datastream build/ssg-rhel8-ds.xml rhelh-vpp
 tests/test_suite.py profile --libvirt qemu:///system test-suite-vm --datastream build/ssg-rhel8-ds.xml rhelh-stig
 tests/test_suite.py profile --libvirt qemu:///system test-suite-vm --datastream build/ssg-rhel8-ds.xml cui
 tests/test_suite.py profile --libvirt qemu:///system test-suite-vm --datastream build/ssg-rhel8-ds.xml ospp

File: linux_os/guide/system/bootloader-zipl/zipl_bls_entries_only/rule.yml [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+description: |-
+    Ensure that zIPL boot entries fully adheres to Boot Loader Specification (BLS)
+    by checking that <tt>/etc/zipl.conf</tt> doesn't contain <tt>image = </tt>.
+
+rationale: |-
+    {{{ full_name }}} adheres to Boot Loader Specification (BLS) and is the prefered method of
+    configuration.
+
+severity: medium
+
```
[the remark this answers is not in the document]
Comment (yuumasato) [reply linux_os/guide/system/bootloader-zipl/zipl_bls_entries_only/rule.yml:16]: CCE added.

File: linux_os/guide/system/bootloader-zipl/zipl_bootmap_is_up_to_date/bash/shared.sh (hunk 3 above)
Comment (yuumasato) [reply linux_os/guide/system/bootloader-zipl/zipl_bootmap_is_up_to_date/bash/shared.sh]: Fixed.

File: linux_os/guide/system/bootloader-zipl/zipl_bls_entries_only/oval/shared.xml (hunk 4 above)
Comment (yuumasato) [reply linux_os/guide/system/bootloader-zipl/zipl_bls_entries_only/oval/shared.xml]: Great finding, the check has been improved.

**Comment by yuumasato** [comment]
Note: references to SFRs to be added later.

**Comment by mildas** [comment]
**Changes identified:**
Rule zipl_bls_entries_only:
 OVAL check is newly added.
Rule zipl_bootmap_is_up_to_date:
 Bash remediation is newly added.
 OVAL check is newly added.
 Ansible remediation newly added.
Profile ospp on rhel8:
 Rule zipl_bootmap_is_up_to_date, zipl_bls_entries_only added to ospp profile.
Profile cui on rhel8:
 CUI profile extends changed OSPP profile.
Profile stig on rhel8:
 Rule !zipl_bls_entries_only, !zipl_bootmap_is_up_to_date added to stig profile.
 STIG profile extends changed OSPP profile.
Profile rhelh-stig on rhel8:
 RHELH-STIG profile extends changed STIG profile.
Profile rhelh-vpp on rhel8:
 RHELH-VPP profile extends changed OSPP profile.
Profile ospp-mls on rhel8:
 OSPP-MLS profile extends changed OSPP profile.
Profile ospp on tests:
 Rule zipl_bootmap_is_up_to_date, zipl_bls_entries_only added to ospp profile.

**Recommended tests to execute:**
 build_product rhel8
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using bash --datastream build/ssg-rhel8-ds.xml zipl_bls_entries_only
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using ansible --datastream build/ssg-rhel8-ds.xml zipl_bootmap_is_up_to_date
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using bash --datastream build/ssg-rhel8-ds.xml zipl_bootmap_is_up_to_date
 tests/test_suite.py profile --libvirt qemu:///system test-suite-vm --datastream build/ssg-rhel8-ds.xml cui
 tests/test_suite.py profile --libvirt qemu:///system test-suite-vm --datastream build/ssg-rhel8-ds.xml ospp-mls
 tests/test_suite.py profile --libvirt qemu:///system test-suite-vm --datastream build/ssg-rhel8-ds.xml rhelh-vpp
 tests/test_suite.py profile --libvirt qemu:///system test-suite-vm --datastream build/ssg-rhel8-ds.xml ospp
 tests/test_suite.py profile --libvirt qemu:///system test-suite-vm --datastream build/ssg-rhel8-ds.xml rhelh-stig
 tests/test_suite.py profile --libvirt qemu:///system test-suite-vm --datastream build/ssg-rhel8-ds.xml stig
 build_product tests
 tests/test_suite.py profile --libvirt qemu:///system test-suite-vm --datastream build/ssg-tests-ds.xml ospp

**Comment by vojtapolasek** [comment]
Please update the STIG stable profile as well.

**Comment by yuumasato** [comment]
> Please update the STIG stable profile as well.

I actually unselected the zIPL rules from the STIG Profile.
33bae25 is going through Jenkins now.

**Comment by matejak** [comment]
Thank you for the PR, merging!

**Comment by yuumasato** [comment]
Thanks for the reviews, :)

**Pull request merged**